File: analytics/src/main/java/io/appmetrica/analytics/impl/AppMetricaServiceCoreImplFieldsFactory.java
package io.appmetrica.analytics.impl;

import android.content.Context;
import androidx.annotation.NonNull;
import io.appmetrica.analytics.coreapi.internal.backport.Consumer;
import io.appmetrica.analytics.impl.component.clients.ClientRepository;
import io.appmetrica.analytics.impl.crash.jvm.CrashDirectoryWatcher;
import java.io.File;

public class AppMetricaServiceCoreImplFieldsFactory {

    public CrashDirectoryWatcher createCrashDirectoryWatcher(
            @NonNull File crashDirectory, @NonNull Consumer<File> newCrashListener
    ) {
        return new CrashDirectoryWatcher(crashDirectory, newCrashListener);
    }

    public ReportConsumer createReportConsumer(@NonNull Context context, @NonNull ClientRepository clientRepository) {
        return new ReportConsumer(context, clientRepository);
    }
}

File: analytics/src/main/java/io/appmetrica/analytics/impl/AppMetricaServiceCoreImplFieldsFactory.kt
package io.appmetrica.analytics.impl

import android.content.Context
import io.appmetrica.analytics.impl.component.clients.ClientRepository

internal class AppMetricaServiceCoreImplFieldsFactory {

    fun createReportConsumer(context: Context, clientRepository: ClientRepository): ReportConsumer {
        return ReportConsumer(context, clientRepository)
    }
}
